stock_picking_filter_lot/models/stock_move_line.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2021 Le Filament (https://le-filament.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import json
+
+from odoo import api, fields, models
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    picking_type_use_filter_lots = fields.Boolean(
+        related="picking_id.picking_type_id.use_filter_lots", readonly=True
+    )
+    lot_id_domain = fields.Char(
+        compute="_compute_lot_id_domain",
+        readonly=True,
+        store=False,
+    )
+
+    @api.depends("picking_type_use_filter_lots")
+    def _compute_lot_id_domain(self):
+        for rec in self:
+            if rec.picking_type_use_filter_lots:
+                rec.lot_id_domain = json.dumps(
+                    [
+                        ("product_id", "=", rec.product_id.id),
+                        ("company_id", "=", rec.company_id.id),
+                        ("location_ids", "child_of", rec.location_id.id),
+                    ]
+                )
+            else:
+                rec.lot_id_domain = json.dumps(
+                    [
+                        ("product_id", "=", rec.product_id.id),
+                        ("company_id", "=", rec.company_id.id),
+                    ]
+                )
```

Comment:
Ok I understand the topic here.
Please you may avoid to have these duplicates lines by adding conditionnally ("location_ids", "child_of", rec.location_id.id) to first 2 tuples and send to json.dumps